### PR TITLE
determine correct Thumb/Thumb2 instruction size (16/32-bit) for code …

### PR DESCRIPTION
**Pull request by 0xSeb**
…hook

### CODE REVIEW

File: qemu/target-arm/translate.c [hunk 1]
```diff
@@ -10407,7 +10407,20 @@ static void disas_thumb_insn(CPUARMState *env, DisasContext *s) // qq
 
     // Unicorn: trace this instruction on request
     if (HOOK_EXISTS_BOUNDED(s->uc, UC_HOOK_CODE, s->pc)) {
-        gen_uc_tracecode(tcg_ctx, 2, UC_HOOK_CODE_IDX, s->uc, s->pc);
+        // determine instruction size (Thumb or Thumb2)
+        insn = arm_lduw_code(env, s->pc, s->bswap_code);
```
Comment (aquynh) [inline qemu/target-arm/translate.c]: can you optimize this code, so we can avoid this calculation again on the line 10428?

**Comment by 0xSeb** [comment]
Agreed, I just did it... I hope it worked I'm new to git-things!

File: qemu/target-arm/translate.c (hunk 1 above)
Comment (0xSeb) [inline qemu/target-arm/translate.c]: done, hopefully it worked?

**Comment by aquynh** [comment]
merged, thanks!

**Pull request merged**